cmd/config-host-list.go

```diff
@@ -0,0 +1,143 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+var configHostListCmd = cli.Command{
+	Name:            "list",
+	ShortName:       "ls",
+	Usage:           "Lists all hosts or a host in mc configuration file.",
+	Action:          mainConfigHostList,
+	Before:          setGlobalsFromContext,
+	Flags:           globalFlags,
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [ALIAS]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. List all hosts.
+     $ {{.HelpName}}
+
+  2. List a specific host.
+     $ {{.HelpName}} s3
+`,
+}
+
+func checkConfigHostListSyntax(ctx *cli.Context) {
+	args := ctx.Args()
+
+	if len(ctx.Args()) > 1 {
+		fatalIf(errInvalidArgument().Trace(args...),
+			"Incorrect number of arguments to list hosts.")
+	}
+
+	if args.Get(0) != "" {
+		if !isValidAlias(args.Get(0)) {
+			fatalIf(errDummy().Trace(args.Get(0)),
+				"Invalid alias `"+args.Get(0)+"`.")
+		}
+	}
+}
+
+func mainConfigHostList(ctx *cli.Context) error {
+	checkConfigHostListSyntax(ctx)
+
+	// Additional command speific theme customization.
+	console.SetColor("HostMessage", color.New(color.FgGreen))
+	console.SetColor("Alias", color.New(color.FgCyan, color.Bold))
+	console.SetColor("URL", color.New(color.FgCyan))
+	console.SetColor("AccessKey", color.New(color.FgBlue))
+	console.SetColor("SecretKey", color.New(color.FgBlue))
+	console.SetColor("API", color.New(color.FgYellow))
+
+	args := ctx.Args()
+	listHosts(args.Get(0)) // List all configured hosts.
+	return nil
+}
+
+// Prints all the hosts.
+func printHosts(maxAlias int, hosts ...hostMessage) {
+	for _, host := range hosts {
+		// Format properly for alignment based on alias length.
+		host.Alias = fmt.Sprintf("%-*.*s", maxAlias, maxAlias, host.Alias)
+		printMsg(host)
+	}
+}
+
+// byAlias is a collection satisfying sort.Interface
+type byAlias []hostMessage
+
+func (d byAlias) Len() int           { return len(d) }
+func (d byAlias) Swap(i, j int)      { d[i], d[j] = d[j], d[i] }
+func (d byAlias) Less(i, j int) bool { return d[i].Alias < d[j].Alias }
+
+// listHosts - list all host URLs or a requested host.
+func listHosts(alias string) {
+	conf, err := loadMcConfig()
+	fatalIf(err.Trace(globalMCConfigVersion), "Unable to load config version `"+globalMCConfigVersion+"`.")
+
+	var maxAlias = 0
+	var hosts []hostMessage
+	for k := range conf.Hosts {
+		if len(k) > maxAlias {
+			maxAlias = len(k)
+		}
+	}
+
+	for k, v := range conf.Hosts {
+		// If specific alias is requested, only print the requested one.
+		if alias != "" && alias == k {
```

Comment:
I think it is more readable if we move out this block outside of for, besides it is better to print an error message when alias is specified but not found

```
if alias != "" {
  if v, ok := conf.Hosts[alias] && ok {
    hosts = []hostsMessage{{  .... }}
  } else {
     error("no alias is found")
  }
}
```